### PR TITLE
Redis rewrite

### LINKED ISSUE #1831: Rewrite of the Redis integration articles
---
[Associated WorkItem - 327370](https://dev.azure.com/msft-skilling/Content/_workitems/edit/327370)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Redis rewrite.

Fixes #1831

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/caching/stackexchange-redis-integration.md](https://github.com/dotnet/docs-aspire/blob/f49edc0cb192e96c887349fca710cc18df3a40c1/docs/caching/stackexchange-redis-integration.md) | [.NET Aspire Redis&reg;<sup>**[*](#registered)**</sup> integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/caching/stackexchange-redis-integration?branch=pr-en-us-1835) |
| [docs/messaging/rabbitmq-integration.md](https://github.com/dotnet/docs-aspire/blob/f49edc0cb192e96c887349fca710cc18df3a40c1/docs/messaging/rabbitmq-integration.md) | [.NET Aspire RabbitMQ integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/rabbitmq-integration?branch=pr-en-us-1835) |
| [docs/whats-new/dotnet-aspire-9-release-candidate-1.md](https://github.com/dotnet/docs-aspire/blob/f49edc0cb192e96c887349fca710cc18df3a40c1/docs/whats-new/dotnet-aspire-9-release-candidate-1.md) | [.NET Aspire 9.0 (Release Candidate 1)](https://review.learn.microsoft.com/en-us/dotnet/aspire/whats-new/dotnet-aspire-9-release-candidate-1?branch=pr-en-us-1835) |

### CODE REVIEW

File: docs/caching/includes/valkey-app-host.md [hunk 1]
```diff
@@ -15,19 +19,88 @@ dotnet add package Aspire.Hosting.Valkey
...
+var cache = builder.AddValkey("cache")
+                   .WithDataVolume(isReadOnly: false);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(cache);
+
+// After adding all resources, run the app...
 ```
+
+The data volume is used to persist the Valkey data outside the lifecycle of its container. The data volume is mounted at the `/data` path in the Valkey container and when a `name` parameter isn't provided, the name is generated at random. For more information on data volumes and details on why they're preferred over [bind mounts](#add-Valkey-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
```
Comment (IEvangelist) [inline docs/caching/includes/valkey-app-host.md:62]:
```suggestion
The data volume is used to persist the Valkey data outside the lifecycle of its container. The data volume is mounted at the `/data` path in the Valkey container and when a `name` parameter isn't provided, the name is generated at random. For more information on data volumes and details on why they're preferred over [bind mounts](#add-valkey-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
```

File: docs/caching/includes/valkey-app-host.md [hunk 2]
```diff
@@ -15,19 +19,88 @@ dotnet add package Aspire.Hosting.Valkey
...
+                       interval: TimeSpan.FromMinutes(5),
+                       keysChangedThreshold: 100);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(cache);
+
+// After adding all resources, run the app...
+```
+
+The preceding code adds persistence to the Redis resource by taking snapshots of the Valkey data at a specified interval and threshold. The `interval` is time between snapshot exports and the `keysChangedThreshold` is the number of key change operations required to trigger a snapshot. For more information on persistence, see [Redis docs: Persistence](https://redis.io/topics/persistence).
```
Comment (IEvangelist) [inline docs/caching/includes/valkey-app-host.md:106]:
```suggestion
The preceding code adds persistence to the Redis resource by taking snapshots of the Valkey data at a specified interval and threshold. The `interval` is time between snapshot exports and the `keysChangedThreshold` is the number of key change operations required to trigger a snapshot. For more information on persistence, see [Redis docs: Persistence](https://redis.io/topics/persistence).

```

File: docs/caching/includes/garnet-app-host.md [hunk 3]
```diff
@@ -15,19 +19,88 @@ dotnet add package Aspire.Hosting.Garnet
...
+```
+
+[!INCLUDE [data-bind-mount-vs-volumes](../../includes/data-bind-mount-vs-volumes.md)]
+
+Data bind mounts rely on the host machine's filesystem to persist the Garnet data across container restarts. The data bind mount is mounted at the `C:\Garnet\Data` on Windows (or `/Garnet/Data` on Unix) path on the host machine in the Garnet container. For more information on data bind mounts, see [Docker docs: Bind mounts](https://docs.docker.com/engine/storage/bind-mounts).
+
+### Add Garnet resource with persistence
+
+To add persistence to the Garnet resource, call the <xref:Aspire.Hosting.GarnetBuilderExtensions.WithPersistence*>
+method with either the data volume or data bind mount:
```
Comment (adegeo) [inline docs/caching/includes/garnet-app-host.md:89]: Accidental sentence split.
```suggestion
To add persistence to the Garnet resource, call the <xref:Aspire.Hosting.GarnetBuilderExtensions.WithPersistence*> method with either the data volume or data bind mount:
```

File: docs/caching/includes/redis-intro.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+ms.topic: include
+---
+
+[Redis](https://redis.io/) is the world's fastest data platform for caching, vector search, and NoSQL databases. The .NET Aspire Redis integration enables you to connect to existing Redis instances, or create new instances from .NET with the [`docker.io/library/redis` container image](https://hub.docker.com/_/redis/).
```
Comment (adegeo) [inline docs/caching/includes/redis-intro.md:5]: To me, this sounds like a marketing statement that might be proven, especially from a docs standpoint. I would just reword that to be less absolute. https://azure.microsoft.com/en-us/products/cache/ has some statements you might be able to use without saying specific that it's the "world's fastest."

> Superior throughput and performance to handle millions of requests per second with down to sub-millisecond latency

> Add a quick caching layer to the application architecture to handle thousands of simultaneous users with near-instant speed—all with the benefits of a fully managed service.

Comment (IEvangelist) [reply docs/caching/includes/redis-intro.md:5]: It's intentional, from their marketing page.

**Pull request merged**